assets/js/phoenix_live_view.js

```diff
@@ -1352,8 +1352,17 @@ class DOMPatch {
         },
         onBeforeNodeDiscarded: (el) => {
           if(el.getAttribute && el.getAttribute(PHX_REMOVE) !== null){ return true }
-          if(el.parentNode !== null && DOM.isPhxUpdate(el.parentNode, phxUpdate, ["append", "prepend"]) && el.id){ return false }
           if(this.skipCIDSibling(el)){ return false }
+          if(el.parentNode !== null && DOM.isPhxUpdate(el.parentNode, phxUpdate, ["append", "prepend"]) && el.nodeType == Node.ELEMENT_NODE){
```

Comment:
Don't we want to discard all nodes, including text ones, except the ones that have IDs?

Reply:
I think this will do that -- as we'll be returning `true` from `onBeforeNodeDiscarded` except for elements with ID.

This will only be called on nodes that morphdom thinks are extra, which isn't necessarily the ones we do, e.g.

if your're going from
```html
\n
<div id="1">
\n
<div id="2">
\n
```

to 

```
\n
<div id="1">
\n
```

morphdom is going to think we want to discard the `<div id="2">` and one of the `\n`'s, and keep the other two.

If we want to actually clean the children that don't have id's, maybe we do something in `onBeforeElUpdated` and clean out any non-ided children. Do you think that's better?

Reply:
> If we want to actually clean the children that don't have id's, maybe we do something in onBeforeElUpdated and clean out any non-ided children. Do you think that's better?

@mveytsman I am thinking that could potentially be better because we will effectively discard all "garbage" and make the diffing/merge itself cleaner. Maybe worth giving it a try in a separate PR?

Reply:
I can don't mind doing it in this one as it's related!

Reply:
It is more so we can compare both approaches side by side but whatever you prefer! :D